addons/mrp/models/mrp_workorder.py

```diff
@@ -503,8 +503,8 @@ def _plan_workorder(self, replan=False):
         # Plan only suitable workorders
         if self.state not in ['pending', 'waiting', 'ready']:
             return
-        if self.date_planned_start:
-            if replan:
+        if self.leave_id:
+            if replan or (self.blocked_by_workorder_ids and start_date > self.leave_id.date_from):
```

Comment:
A bit too magical for me. I would not erase the user input. Don't do anything. The warning icon will be display in the list view and the user could use it to replan everything correctly